File: app/src/main/java/com/alexkn/syntact/dataaccess/dao/ClueDao.java
package com.alexkn.syntact.dataaccess.dao;

import androidx.room.Dao;
import androidx.room.Query;

import com.alexkn.syntact.dataaccess.dao.base.BaseDao;
import com.alexkn.syntact.domain.model.Clue;

@Dao
public interface ClueDao extends BaseDao<Clue> {

    @Query("SELECT * FROM Clue where clueId = :id")
    Clue find(Long id);
}

File: app/src/main/java/dev/alexknittel/syntact/data/dao/ClueDao.kt
package dev.alexknittel.syntact.data.dao

import androidx.room.Dao
import androidx.room.Query
import dev.alexknittel.syntact.core.model.Clue
import dev.alexknittel.syntact.data.dao.base.BaseDao

@Dao
interface ClueDao : BaseDao<Clue> {

    @Query("SELECT * FROM Clue where clueId = :id")
    suspend fun find(id: Long): Clue
}
